### PR TITLE
[UI] Admin portal Initial validation error fix

### PR DESCRIPTION
**Pull request by chathuranga95**
### Description

On the Admin Portal, when a Add popup is displayed, errors will be shown on mandatory fields (Field is empty).
**Expected behavior:** Validation should start after the user has started entering values.

This PR fixes the mentioned behavior.

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/AdminPages/Addons/InputListBase.jsx [hunk 1]
```diff
@@ -56,13 +56,13 @@ const InputList = (props) => {
...
             setId(id + 1);
-            setUserInputItems([{ key: '' + id, value: '', error: onValidation('') }]);
+            setUserInputItems([{ key: '' + id, value: '', error: onValidation(undefined) }]);
         }
     }, []);
 
     const onAddInputField = () => {
         setId(id + 1);
-        const newUserItemList = [...userInputItems, { key: '' + id, value: '', error: onValidation('') }];
+        const newUserItemList = [...userInputItems, { key: '' + id, value: '', error: onValidation(undefined) }];
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/AdminPages/Addons/InputListBase.jsx:65]: shall we avoid passing using these `undefined` values here? and other places as well

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/APICategories/AddEditAPICategory.jsx [hunk 2]
```diff
@@ -56,7 +56,7 @@ function AddEdit(props) {
 
     let id = null;
     let initialState = {
-        name: '',
+        name: undefined,
```
[the remark this answers is not in the document]
Comment (chathuranga95) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/APICategories/AddEditAPICategory.jsx]: Fixed with commit: https://github.com/wso2/carbon-apimgt/pull/8838/commits/2455d6596ee3933b261b5d8897ae2f37e49fd66e

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/AdminPages/Addons/InputListBase.jsx (hunk 1 above)
Comment (chathuranga95) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/AdminPages/Addons/InputListBase.jsx]: Fixed with commit: https://github.com/wso2/carbon-apimgt/pull/8838/commits/2455d6596ee3933b261b5d8897ae2f37e49fd66e

**Comment by chathuranga95** [comment]
Related Issue: https://github.com/wso2/product-apim/issues/8408

**Pull request merged**